ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -225,14 +225,14 @@ void registerModel(ActionListener<MLRegisterModelResponse> listener) throws Inte
             XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
             modelGroup.toXContent(builder, ToXContent.EMPTY_PARAMS);
             createModelGroupRequest.source(builder);
-            client.index(createModelGroupRequest, ActionListener.wrap(r -> {
+            client.index(createModelGroupRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
```

Comment:
May be we should add a similar comment in the code also what you have in the PR?